# Environments

## Environment 1 - L4137_01_Protease

**List of software included in environment**

| Software |  Version | 
| -------- | -------- | 
| FastQC    |   0.12.1   | 
| miRTrace    |   1.0.1   | 
| MultiQC    |   1.34   | 





## Environment 2 - L4137_01_Protease_cutadapt

**List of software included in environment**

| Software |  Version | 
| -------- | -------- | 
| cutadapt    |   5.2   | 





## Environment 3 - L4137_01_Protease_bowtie

**List of software included in environment**

| Software |  Version | 
| -------- | -------- | 
| bowtie    |   1.3.1   | 
| samtools    |   1.23.1   | 





## Environment 4 - L4137_01_Protease_mirdeep2

**List of software included in environment**

| Software |  Version | 
| -------- | -------- | 
| miRDeep2    |   2.0.1.3   | 





# Input Data

- Single-end reads
- One FASTQ file per sample 
- Single lane (merging multiple lanes not required)

In [1]:
library(readr)
X01_sample_metadata <- read_csv("/Users/christopherjanschke/GIT/L4137_HBEC_EV_Transcriptomics-Project/01_Protease/results/tables/01_sample_metadata.csv", show_col_types = FALSE)
X01_sample_metadata

Warning message:
“package ‘readr’ was built under R version 4.5.2”


Condition,Treatment,Replicate,Sequencing ID,File Size
<chr>,<chr>,<dbl>,<chr>,<chr>
PBS_03,Untreated,1,NGS-110-028_S28_R1_001,354M
PBS_04,Untreated,2,NGS-110-016_S16_R1_001,374M
PBS_06,Untreated,3,NGS-110-007_S7_R1_001,282M
0_005_ug_mL_Protease_01,Treated,1,NGS-110-029_S29_R1_001,318M
0_005_ug_mL_Protease_11,Treated,2,NGS-110-017_S17_R1_001,365M
0_005_ug_mL_Protease_14,Treated,3,NGS-110-011_S11_R1_001,373M
0_08_ug_mL_Protease_13,Treated,1,NGS-110-012_S12_R1_001,371M
0_08_ug_mL_Protease_08,Treated,2,NGS-110-025_S25_R1_001,397M
0_08_ug_mL_Protease_09,Treated,3,NGS-110-006_S6_R1_001,314M


# Processing Steps

## 01. Data Preparation

### Final

### Test

#### 1. Input Data

**Script name:** [].sh 

**1. Description** 

**2. Software** 
N/A

**3. Software Justification**
Re-naming files is a simple in-built Linux capability - no specialist software required.

**4. Results**
Files named according to a standardised format to be used as input data for downstream scripts.

**5. Notes**
- Takes in file paths, re-names according to convention then saves them in `raw_data`; .csv containing metadata for traceability is produced. 
- Requires single-end read data
- Wildcards (e.g., `*`, `?`) required if files from multiple lanes exist. The script will concatenate multiple files into one merged file.

**Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |
| `[]` | `[]` | `[]`  | [] |




## 02. Quality Control

### Final

### Test

#### Software 1: FastQC

**Script name:** 
02a_raw_reads.sh 
02c_trimmed_reads.sh

**1. Description** 
Produces a quality control report for each individual sample. Quality control metrics assessed:
- Basic Statistics
- Per base sequence quality
- Per tile sequence quality
- Per sequence quality scores
- Per base sequence content
- Per sequence GC content
- Per base N content
- Sequence Length Distribution
- Sequence Duplication Levels
- Overrepresented sequences
- Adapter Content

**2. Software** 
[FastQC](https://github.com/s-andrews/FastQC) v0.12.1

**3. Justification**
FastQC is a well-established and reliable quality control software used in multi-omics analyses.

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |
| `NGS-110-<SAMPLE_NUMBER>_S<SAMPLE_NUMBER>_R1_001` | `fastq.gz` | `[]`  | Raw gzipped RNA-Seq FASTQ files containing single-read data for each sample. Name description (from left to right): `NGS` = Next Generation Sequencing, `110` = [], `<SAMPLE_NUMBER>` is replaced with the appropriate sample number (001-030), `S<SAMPLE_NUMBER>` where `<SAMPLE_NUMBER>` is replaced with sample number (01-30 - note: only double-digit), `R1` = single-read, `001` = sequencing lane 1 |
| `NGS-110-<SAMPLE_NUMBER>_S<SAMPLE_NUMBER>_R1_001_trimmed` | `fastq.gz` | `[]`  | Raw gzipped RNA-Seq FASTQ files containing single-read data for each sample. Name description (from left to right): `NGS` = Next Generation Sequencing, `110` = [], `<SAMPLE_NUMBER>` is replaced with the appropriate sample number (001-030), `S<SAMPLE_NUMBER>` where `<SAMPLE_NUMBER>` is replaced with sample number (01-30 - note: only double-digit), `R1` = single-read, `001` = sequencing lane 1, `trimmed` = the trimmed reads (after adapter contamination is removed) |

**5a. Results - RAW DATA**

#### Table containing general statistics
| Sample Name              | Dups  | GC    | Seqs |
|--------------------------|-------|-------|------|
| NGS-110-003_S3_R1_001   | 83.2% | 65.0% | 9.4M |
| NGS-110-004_S4_R1_001   | 82.1% | 63.0% | 9.9M |
| NGS-110-006_S6_R1_001   | 81.8% | 64.0% | 8.3M |
| NGS-110-007_S7_R1_001   | 80.5% | 64.0% | 7.8M |
| NGS-110-008_S8_R1_001   | 88.4% | 66.0% | 10.8M |
| NGS-110-011_S11_R1_001  | 83.7% | 63.0% | 10.3M |
| NGS-110-012_S12_R1_001  | 82.1% | 65.0% | 9.8M |
| NGS-110-016_S16_R1_001  | 81.8% | 63.0% | 9.6M |
| NGS-110-017_S17_R1_001  | 83.3% | 62.0% | 10.0M |
| NGS-110-019_S19_R1_001  | 87.3% | 64.0% | 10.1M |
| NGS-110-025_S25_R1_001  | 84.0% | 62.0% | 10.5M |
| NGS-110-026_S26_R1_001  | 83.6% | 65.0% | 10.3M |
| NGS-110-028_S28_R1_001  | 82.6% | 65.0% | 10.1M |
| NGS-110-029_S29_R1_001  | 82.3% | 64.0% | 9.0M |
| NGS-110-030_S30_R1_001  | 86.2% | 65.0% | 10.2M |


#### Table containing the overrepresented sequences 

- Reports = number of individual samples with the sequence (15 samples total)
- Occurrences = number of times this sequence appeared
- `TGGAATTCTCGG` appears in all overrepresented sequences - highlight in bold
- TGGAATTCTCGG is the Illumina Small RNA 3' Adapter [reference](https://bioinformatics.ccr.cancer.gov/docs/b4b/10.-Removing-adapters-from-sequence-data%20copy/)

| Overrepresented sequence                                | Reports | Occurrences | % of all reads |
| ------------------------------------------------------- | ------: | ----------: | -------------: |
| ATCCCTGGTGGTCTAGTGGTTAGGATTCGGCGC**TGGAATTCTCGG**GTGCC  |      14 |     1825375 |          1.25% |
| ATGGAATTCTCGGGTGCCAAGG**TGGAATTCTCGG**GTGCCAAGGAACTCCA  |      15 |     1238413 |          0.85% |
| AGGAGAAGCCGGCGGGAGCCCCGGGGAGS**TGGAATTCTCGG**GTGCCAAGGA |      11 |     1080775 |          0.74% |
| ATCCCTGGTGGTCTAGTGGTTAGGATTCGGCG**TGGAATTCTCGG**GTGCCA  |      12 |      790317 |          0.54% |
| AGGAGAAGCCGGCGGGAGCCCCGGGGAGA**GTGGAATTCTCGG**GTGCCAAG  |      12 |      715410 |          0.49% |
| ATCCCTGGTGGTCTAGTGGTTAGGATTCGGCGCT**TGGAATTCTCGG**GTGC  |       7 |      643902 |          0.44% |
| AGGAGAAGCCGGCGGGAGCCCCGGGG**TGGAATTCTCGG**GTGCCAAGGAAC  |      11 |      637188 |          0.44% |
| AGTACTCGGCTTACGATCGTAAT**GGAATTCTCGG**GTGCCAAGGAACTCCA  |      15 |      573609 |          0.39% |
| AGGTAGCATGCTTGTCTGAGTGTCTGT**TGGAATTCTCGG**GTGCCAAGGAA  |      12 |      230359 |          0.16% |
| AGCATTGGTGGTTCAGTGGTAGAATTCTCGC**TGGAATTCTCGG**GTGCCAA  |       8 |      228972 |          0.16% |
| AAGGTTCAGCCTCTGTTCCGGGATTTGT**TGGAATTCTCGG**GTGCCAAGGA  |      12 |      188149 |          0.13% |
| ACGGTAGCATGCTTGTCTGAGTGTCTGT**TGGAATTCTCGG**GTGCCAAGGA  |      12 |      186029 |          0.13% |
| AGGAGAAGCCGGCGGGAGCCCCGGG**TGGAATTCTCGG**GTGCCAAGGAACT  |       6 |      173797 |          0.12% |
| AAGCATGCTTGTCTGAGTGTCTGT**TGGAATTCTCGG**GTGCCAAGGAACTC  |      12 |      167406 |          0.11% |
| AGCATGCTTGTCTGAGTGTCTGT**TGGAATTCTCGG**GTGCCAAGGAACTCC  |      11 |      139135 |          0.10% |
| AGCCCGGATAGCTCAGTCGGTAGAGCATCAGAC**TGGAATTCTCGG**GTGCC  |       6 |      119906 |          0.08% |
| ATGCTGAGGTTCAGCCTCTGTTCCGGGATTTGT**TGGAATTCTCGG**GTGCC  |      10 |      114791 |          0.08% |
| ATGACGGTCGTGCGACCATTCGT**TGGAATTCTCGG**GTGCCAAGGAACTCC  |      10 |      111887 |          0.08% |
| AAGCCTCTGTTCCGGGATTTGT**TGGAATTCTCGG**GTGCCAAGGAACTCCA  |       9 |      102972 |          0.07% |
| AGGTTCAGCCTCTGTTCCGGGATTTGT**TGGAATTCTCGG**GTGCCAAGGAA  |       8 |       93323 |          0.06% |


**5b. Results - TRIMMED DATA**

#### Table containing general statistics

| Sample Name               | Dups  | GC    | Seqs |
|--------------------------|-------|-------|------|
| NGS-110-003_S3_R1_001    | 86.3% | 49.0% | 5.7M |
| NGS-110-004_S4_R1_001    | 85.7% | 49.0% | 6.0M |
| NGS-110-006_S6_R1_001    | 85.6% | 49.0% | 5.2M |
| NGS-110-007_S7_R1_001    | 84.6% | 48.0% | 4.9M |
| NGS-110-008_S8_R1_001    | 88.7% | 62.0% | 5.2M |
| NGS-110-011_S11_R1_001   | 87.8% | 50.0% | 6.5M |
| NGS-110-012_S12_R1_001   | 86.3% | 49.0% | 6.0M |
| NGS-110-016_S16_R1_001   | 86.1% | 48.0% | 6.0M |
| NGS-110-017_S17_R1_001   | 87.3% | 49.0% | 6.4M |
| NGS-110-019_S19_R1_001   | 87.9% | 61.0% | 5.1M |
| NGS-110-025_S25_R1_001   | 87.9% | 49.0% | 6.5M |
| NGS-110-026_S26_R1_001   | 86.5% | 49.0% | 6.0M |
| NGS-110-028_S28_R1_001   | 87.1% | 48.0% | 6.3M |
| NGS-110-029_S29_R1_001   | 84.8% | 48.0% | 5.0M |
| NGS-110-030_S30_R1_001   | 87.4% | 59.0% | 5.4M |

#### Table containing the overrepresented sequences 
- Reports = number of individual samples with the sequence (15 samples total)
- Occurrences = number of times this sequence appeared
- These sequences do not correspond to known adapters - they are likely real miRNA sequences which occur many times in the samples.

| Overrepresented sequence            | Reports | Occurrences | % of all reads |
|-------------------------------------|---------|-------------|----------------|
| AGGAGAAGCCGGCGGGAGCCCCGGGGAG        | 14      | 1181270     | 1.3724%        |
| AGGAGAAGCCGGCGGGAGCCCCGGGGAGAG      | 14      | 775650      | 0.9012%        |
| AGGAGAAGCCGGCGGGAGCCCCGGGGAGA       | 11      | 764090      | 0.8878%        |
| AGGAGAAGCCGGCGGGAGCCCCGGGG          | 14      | 710866      | 0.8259%        |
| AGTACTCGGCTTACGATCGTAA              | 15      | 680152      | 0.7902%        |
| AGGTAGCATGCTTGTCTGAGTGTCTGT         | 15      | 270269      | 0.3140%        |
| AAGGTTCAGCCTCTGTTCCGGGATTTGT        | 13      | 208074      | 0.2417%        |
| ACGGTAGCATGCTTGTCTGAGTGTCTGT        | 13      | 206964      | 0.2405%        |
| AAGCATGCTTGTCTGAGTGTCTGT            | 13      | 192010      | 0.2231%        |
| AGCATGCTTGTCTGAGTGTCTGT             | 12      | 167815      | 0.1950%        |
| ATGACGGTCGTGCGACCATTCGT             | 13      | 151044      | 0.1755%        |
| AAGCCTCTGTTCCGGGATTTGT              | 12      | 148901      | 0.1730%        |
| AGGTTCAGCCTCTGTTCCGGGATTTGT         | 12      | 139486      | 0.1621%        |
| AATGCTTTCGAGTGAAACGGATT             | 12      | 118486      | 0.1377%        |
| ATGCGGAAGGATCATTG                   | 12      | 110343      | 0.1282%        |
| AGCTTTCGAGTGAAACGGATT               | 12      | 105439      | 0.1225%        |
| ACGGGAAGTGCTCAGGATGTGAATCTGCGC      | 12      | 105316      | 0.1224%        |
| ATGCTTTCGAGTGAAACGGATT              | 11      | 88308       | 0.1026%        |
| AAGGTTCAGCCCGTGTTTCAAGATTTGT        | 10      | 78311       | 0.0910%        |
| AGTAGCATGCTTGTCTGAGTGTCTGT          | 10      | 69018       | 0.0802%        |

#### Sequence Length Distribution
- All reads within 17-30bp in length (this aligns with trimming parameters set at min=17 and max=30).
- Highest number of reads at 22bp.
- Less reads observed at the lower length range (e.g., 17-20).
- All samples show similar trend in length distribution except for three samples which have a slightly different distribution (but the same as each other):
1. Sample 8
2. Sample 19
3. Sample 30

**N.B. Samples 8, 19 and 30 are all from the 24ug (highest concentration) condition.**

![](project_worklog_images/fastqc/fastqc_sequence_length_distribution.png)

#### Sequence Duplication Levels
- All samples failed this based on FastQC's criteria.
- However, for miRNA it is expected to naturally observe duplication for mature miRNAs that are highly abundant, so this is not a concern.


![](project_worklog_images/fastqc/fastqc_sequence_duplication.png)

#### Per Sequence GC Content
- Another instance where FastQC flags some samples as bad, but this is likely, at least in part, due to differences in miRNA sequencing vs typical WGS.
- The raw read data graph is cleaner.
- Most samples show a similar trend (which is the main thing). Three samples don't (but they follow a shared trend), these are:
1. Sample 8
2. Sample 19
3. Sample 30

**N.B. Samples 8, 19 and 30 are all from the 24ug (highest concentration) condition.**


![](project_worklog_images/fastqc/fastqc_per_sequence_GC_content.png)

#### Quality Scores
- Base and Sequence quality scores were all in the green (approx. Phred 35-40).


![](project_worklog_images/fastqc/fastqc_quality_scores.png)

**6. Notes**
- TamiRNA MultiQC report looks different to this trimmed one. This is likely due to TamiRNA not providing a maximum read length whereas this analysis did. 
- See i_120526_01 (`<PROJECT_ROOT>_investigations/`) for more details - the analysis will be repeated without a maximum read length in a later investigation to provide a comparison.




#### Software 2: miRTrace

**Script name:**
02a_raw_reads.sh 
02c_trimmed_reads.sh

**1. Description** 
mirTRace is a tool for quality control and tracing taxonomic origins of microRNA sequencing data.

**2. Software** 
[mirTRace](https://github.com/friedlanderlab/mirtrace) v1.0.1

**3. Justification**
mirTRace is designed specifically for small RNA sequencing. It was included in the TamiRNA MultiQC report (though not specified in the 2022 mIND published paper) where it classified the species of origin for reads.

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |
| `NGS-110-<SAMPLE_NUMBER>_S<SAMPLE_NUMBER>_R1_001` | `fastq.gz` | `[]`  | Raw gzipped RNA-Seq FASTQ files containing single-read data for each sample. Name description (from left to right): `NGS` = Next Generation Sequencing, `110` = [], `<SAMPLE_NUMBER>` is replaced with the appropriate sample number (001-030), `S<SAMPLE_NUMBER>` where `<SAMPLE_NUMBER>` is replaced with sample number (01-30 - note: only double-digit), `R1` = single-read, `001` = sequencing lane 1 |
| `NGS-110-<SAMPLE_NUMBER>_S<SAMPLE_NUMBER>_R1_001_trimmed` | `fastq.gz` | `[]`  | Raw gzipped RNA-Seq FASTQ files containing single-read data for each sample. Name description (from left to right): `NGS` = Next Generation Sequencing, `110` = [], `<SAMPLE_NUMBER>` is replaced with the appropriate sample number (001-030), `S<SAMPLE_NUMBER>` where `<SAMPLE_NUMBER>` is replaced with sample number (01-30 - note: only double-digit), `R1` = single-read, `001` = sequencing lane 1, `trimmed` = the trimmed reads (after adapter contamination is removed) |

**5. Results**
#### QC Plot
- The Majority of reads are without adapters - however this was also seen on the raw data QC plot.
- Some reads < 18nt because the minimum length threshold was set at 17 during trimming. 
- Very small amount (<200) of reads with low complexity. [significance?]
- Very small amount (<2000) of reads with low Phred score.

![](project_worklog_images/mirTRace/mirTRace_QC_plot.png)

#### RNA Categories
- Shows RNA classifications per sample:
1. Unknown (highest)
2. rRNA
3. miRNA
4. tRNA
5. Artifact (lowest)
- For raw reads, 'Artifact' and 'Unknown' were the highest, respectively.

![](project_worklog_images/mirTRace/mirTRace_RNA_categories.png)

#### Read Length Distribution
- All reads are 17-30nt long (trimming settings were set at 17-30nt, so this makes sense).
- Most frequently observed length is 22nt.
- For raw reads, all reads were 50nt long. FastQC said that all reads have "single length (100bp)".

![](project_worklog_images/mirTRace/mirTRace_read_length_distribution.png)

#### Contamination Check
- This section detects clades from a variety of different organisms.
- Clades were detected in only 6 samples - all from primates.

![](project_worklog_images/mirTRace/mirTRace_contamination_check.png)

#### miRNA Complexity
- Shows the number of distinct miRNAs detected in each sample.
- It shows the number detected as sequencing depth increases but take-home message is how many overall.
- Distinct miRNAs ranges from 198-368 across samples.
- For raw data, this doesn't work very well - only detecting approx. one miRNA per sample.

![](project_worklog_images/mirTRace/mirTRace_miRNA_complexity.png)

**6. Notes**
The miRTrace section in this MultiQC report does not look the same as how it appears in the TamiRNA report - TamiRNA






#### Software 3: MultiQC

**Script name:** [].sh 

**1. Description** 
Software to compile individual sample QC reports into one merged report to speed up manual review of quality data.

**2. Software** 
[MultiQC](https://github.com/MultiQC/MultiQC) v1.34

**3. Justification**
- MultiQC is a well-established method to compile multiple QC reports.
- It auto-detects the software that produced the initial QC reports and whether it was pre- or post-trimming.

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |
| `[]` | `[]` | `[]`  | [] |


**5. Results**
See results for miTRace and FastQC.

**6. Notes**
[]





#### Software 4: cutadapt

**Script name:** [].sh 

**1. Description** 
[]

**2. Software** 
[cutadapt](https://github.com/marcelm/cutadapt/) v5.2

**3. Justification**
[]

**4. Results**
[]

**5. Notes**
- Default error rate is 0.1 - Error rate = no. of errors divided by length of matching region.
- Default overlap is 3 - Require MINLENGTH overlap between read and adapter for an adapter to be found.

**Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |
| `[]` | `[]` | `[]`  | [] |




#### Software 5: fastp

**Script name:** [].sh 

**1. Description** 
[]

**2. Software** 
[]

**3. Justification**
[]

**4. Results**
[]

**5. Notes**
- []

**Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |
| `[]` | `[]` | `[]`  | [] |




## 03. Alignment

### Final

### Test

#### 1. Download References

**Script name:** 
[]

**1. Description** 
Download reference files necessary to create bowtie index and perform alignment. References to download:
1. (Ensembl) human genome 
2. (miRBase) mature miRNA sequences
3. (miRBase) hairpin miRNA sequences

Indexing using Samtools not necessary as bowtie will create index files.

**2. Software** 
N/A

**3. Justification**
All commands used are in-built Linux commands - no specialist software required.

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |


**5. Commands**

**Download homo *sapien* reference genome from Ensembl (gzipped FASTA):**

```wget https://ftp.ensembl.org/pub/release-115/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz```

**Download mature miRNA sequences from miRBase (Release 22.1 - none-zipped FASTA):**

```wget https://www.mirbase.org/download/CURRENT/mature.fa```


**Download mature miRNA sequences from miRBase (Release 22.1 - none-zipped FASTA):**

```wget https://www.mirbase.org/download/CURRENT/hairpin.fa```

**6. Results**
[]

**7. Notes**
1. wget does not overwrite a file if it already exists therefore the script will fail. Corrective actions:
- There is no flag to overwrite existing files or check the file size before download (in case of partial download)
- Need to do a check for the file and if found, then delete it (or check file size matches the expected file size and if not, delete it?)
- Then use wget to download it again

2. Need to remove whitespace (included in script) otherwise you get this error from miRDeep2: `Error: Genome file /gpfs01/home/mbxcj2/L4137_HBEC_EV_Transcriptomics-Project/01_Protease/data/reference/GRCh38.p14/Homo_sapiens.GRCh38.dna.primary_assembly.fa has not allowed whitespaces in its first identifier`



#### Software 2: bowtie

**Script name:** 
[]

**1. Description** 
[]

**2. Software** 
[bowtie](https://github.com/BenLangmead/bowtie) v1.3.1

**3. Justification**
Bowtie is old (newer version = bowtie2) however it is very good at analysis short read data. It is still widely used for miRNA analysis.

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |

**5. Commands**
`--norc` = Does not align to reverse-complement strand. This is used because the sequencing was single-stranded.
`v2` = 2 mismatches (other papers use 1 or 0 but TamiRNA used 2)
`a` = reports all locations that each read aligns to e.g. read aligns to multiple loci, all loci are reported
`--best --strata` = report hits from the best stratum

**6. Results**
[]

**7. Notes**
1. Need to perform a check for all the relevant index files (and ensure the files are not empty) and only build the index if they're not present - this saves time re-making the files. Perform the same check after they're amde to ensure they were created properly.




#### Software 3: miRDeep2

**Script name:** 
[]

**1. Description** 
[]

**2. Software** 
[miRDeep2](https://github.com/rajewsky-lab/mirdeep2) v2.0.1.3

**3. Justification**
[]

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |


**5. Commands**
mapper.pl - mapping (collapsed reads)

`-s` = processed reads printed to this file (only one instance of a read is written accompanied by the number of times it appears)
`-t` = read mappings printed to this file (alignment)
`-l 18` = discard reads shorted than 18nt
`-n` = overwrite existing files
`-h` = parse to fasta format
`-e` = input file is in fastq format
`-i` = converts 'U' to 'T' (RNA to DNA)
`-j` = remove entries that have entries other than: a,c,g,t,u,n,A,C,G,T,U,N
`-m` = collapse reads (create one entry but with number indicating its no. of appearances)
`-k TGGAATTCTCGGGTGCCAAGG` = trim adapter sequence




miRDeep2.pl

quantifier.pl - expression matrix


**6. Results**
[]

**7. Notes**
- Documentation [here](https://www.mdc-berlin.de/content/mirdeep2-documentation?mdcbl%5B0%5D=/n-rajewsky%23t-joinus&mdctl=0&mdcou=20738&mdcot=6&mdcbv=5yVgWgdVqnN3ZrjWVN8SYGoN7LptyyVi7YGzkgWdh9E)
- Tutorial [here](https://deepwiki.com/rajewsky-lab/mirdeep2/9-tutorial)






#### Software 4: miRBase

**Script name:** 
[]

**1. Description** 
- Contains well-established miRNA reference files.
- mature.fa = mature miRNA sequences
- hairpin.fa = pre-cursors sequences

**2. Software** 
[miRBase](https://github.com/biobricks-ai/miRBase) v[]

**3. Justification**
[]

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |


**5. Commands**
[]

**6. Results**
[]

**7. Notes**
[]






#### Software 5: FeatureCounts

**Script name:** 
[]

**1. Description** 
[]

**2. Software** 
[FeatureCounts](https://github.com/byee4/featureCounts/blob/master/README.md) v[]

**3. Justification**
[]

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |


**5. Commands**
[]

**6. Results**
[]

**7. Notes**
[]






#### Software 6: HTSeq

**Script name:** 
[]

**1. Description** 
[]

**2. Software** 
[HTSeq](https://github.com/htseq/htseq) v[]

**3. Justification**
[]

**4. Inputs Files**
| Name                           | Extension | File Size | Description                                                                                                                |
| ------------------------------ | --------- | --------- | -------------------------------------------------------------------------------------------------------------------------- |


**5. Commands**
[]

**6. Results**
[]

**7. Notes**
[]




